includes/class-wc-stripe-blocks-support.php

```diff
@@ -224,11 +224,13 @@ private function should_show_payment_request_button() {
 	 * @return array  the JS configuration from the Stripe Payment Gateway.
 	 */
 	private function get_gateway_javascript_params() {
-		$js_configuration = [];
+		$js_configuration   = [];
+		$available_gateways = WC()->payment_gateways->get_available_payment_gateways();
 
-		$gateways = WC()->payment_gateways->get_available_payment_gateways();
-		if ( isset( $gateways['stripe'] ) ) {
-			$js_configuration = $gateways['stripe']->javascript_params();
+		if ( isset( $available_gateways['stripe'] ) ) {
+			$js_configuration = $available_gateways['stripe']->javascript_params();
+		} elseif ( $this->is_upe_method_available( $available_gateways ) ) {
+			$js_configuration = WC_Stripe::get_instance()->get_main_stripe_gateway()->javascript_params();
```

Comment:
Nitting a bit about an edge case here:

1. Enable UPE
2. Disable all of the payment method types from the Stripe > Settings > Payment methods tab
3. Go to the block checkout and open the console
4. Notice there's an error saying 
```
Uncaught TypeError: Cannot convert undefined or null to object
    at entries (<anonymous>)
    at ./client/blocks/upe/index.js (index.js:37:1)
    at __webpack_require__ (upe_blocks.js?ver=50d725475582a9d2698bf6e8d71a443e:20:30)
    at upe_blocks.js?ver=50d725475582a9d2698bf6e8d71a443e:84:18
    at upe_blocks.js?ver=50d725475582a9d2698bf6e8d71a443e:87:10
```

This is coming from [this line](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/add/deferred-intent/client/blocks/upe/index.js#L24). The JS param `paymentMethodsConfig`, defined [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/add/deferred-intent/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L326) isn't provided in this case.

It'd be great not to load those scripts when no payment gateways are enabled, but it's probably better to handle this in a separate issue to keep this PR small.

Reply:
> It'd be great not to load those scripts when no payment gateways are enabled, but it's probably better to handle this in a separate issue to keep this PR small.

Sounds good. :) I'll file an issue for it

Reply:
To close the loop here. Here's the new issue for this: https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2865